lib/replicator.js

```diff
@@ -1673,17 +1673,13 @@ function clampRange (core, r) {
   if (r.blocks === null) {
     const start = core.bitfield.firstUnset(r.start)
 
-    if (r.end === -1) {
-      r.start = start === -1 ? core.tree.length : start
-    } else {
-      const end = core.bitfield.lastUnset(r.end - 1) + 1
-
-      if (start === -1) {
-        r.start = r.end
-      } else {
-        r.start = start
-        r.end = end
-      }
+    if (r.end === -1) r.start = start === -1 ? core.tree.length : start
+    else if (start === -1) r.start = r.end
+    else {
+      const end = core.bitfield.lastUnset(r.end)
+
+      r.start = start
+      r.end = end
```

Comment:
This seems a little more conservative than previously, but would not move `end` if `end - 1` is a set bit.